### ISSUE TITLE
Have Cloaking make shields slightly less efficient

### ISSUE THREAD

**Issue by clifflocked** [endless-sky/endless-sky#5805]
**Describe the solution you'd like**
Currently, cloaking is a bit OP. Yes, only a couple of ships can do it(4 of your choice, and also some of the Remnant ones), but enemies can't target you, allowing you to jump away very easily. I feel as though the cloaking device should cause the shields to not work so that hull damage is the only damage, or a combination of hull & shield damage is done. I feel as though this would make cloaking in Automata space much more interesting, because now you have to not only avoid the fire of most of the ships, but *stay away from the Mereti's shield disruptors*.

**Describe alternatives you've considered**
Maybe the shield generators could generate fewer of the shields for the same amount of power. 

**Additional context**
The "lore" behind this would be that shields and cloaking have to generate their effect in the same place. So, the shields have to generate more shields to make way for the cloaking, and thus produce less shielding per second.

**Comment by gunqqer** [comment]
Cloaking imo is currently endgame stuff so being unbalanced isn't terrible. I think we should wait to try and rebalance the current cloak until we have more types of cloaking (#4119) and/or implement more cloaking related things (#3825)

**Comment by Zitchas** [comment]
You are not alone in thinking that cloaks need a nerf, such as preventing shields from regenerating. That being said, I'm not one of them. Or at least, not until #4119 is resolved. I would very much like to have a variety of different cloaking effects. I would be fine with having *some* cloaks blocking shields, so long as it isn't *all* cloaks.

**Comment by movingpictures** [comment]
I'd argue that for some folks, cloaking is NOT endgame stuff. IN my attempts to break the game for the sake of humanity, I've discovered that a Starling is a hell of an effective alternate flagship. The mission to capture surveillance drone bits is cake with one.

**Comment by gunqqer** [comment]
If you figure out the Remnant before freeworlds, well, that's on you. I don't like balancing for people who are minmaxing and metagaming. Could the starling use a different type of cloak and that'd be better? Yeah. But we shouldn't change the singular type of cloak we currently have just for the Starling but rather introduce new ones.

**Comment by movingpictures** [comment]
Hey, I grasp the whole mantra of not tinkering for those on the edges of
the bell curve - and to be honest, I'm really only trying to push the
boundaries of what's out there as my way of giving  back to a game I've
much enjoyed. Which is why I've delved into asteroid mining, and
intentionally avoiding the F/W line so that i can explore others. (And yes,
try to find things to break for the betterment of humanity.)

**Comment by pepoluan** [comment]
A simple nerf would be "cannot jump / hyper when cloaked".

So when you jump/hyper into a system (say, the Korath AI farming system), you'll have several seconds of sheer terror as you try to do evasive maneuvers while trying desperately to cloak.

Also making the ship more sluggish while cloaking is also a good nerf; it will force players to kinda choose between tradeoff of 'safety' vs 'speed'.

Also a good nerf would be to tie cloaking not only to hyperfuel, but also to battery charge and/or temperature. That will force players to build a ship with larger battery capacity and/or more cooling systems.

Of course, the above can be selectively applied. Say:
* Starling cloaks don't use battery charge and/or cause heat because it's "organic", it only uses hyperfuel
* Korath cloak (that the player receives) uses battery or causes heat or both
* Both of the above mess up with jumping/hypersliding
* Pug cloaks (if we consider this the "god" tier) uses hyperfuel but does not mess up with jumping/hypersliding (because the Pugs, lore-wise, knows what's up with jumpin' and hyperin')

**Comment by Zitchas** [comment]
Yes, those are all good ideas - but only once we have implemented mechanisms for having different kinds of cloaks. Until then, cloaking should remain pegged to the highest user of cloaks (which in this case, is Archons and Pug).

**Comment by Quantumshark** [comment]
Framework for differentiable cloaks has been implemented in #7025, so I guess this is just waiting for someone to code a `cloaked shield permeability` attribute?

**Comment by Quantumshark** [comment]
Since #10610 has been merged, we now have all the necessary code to implement this.
With that in mind, we can now look at the balance side of this in isolation. I agree that cloaking is really quite overpowered currently, so using this new attribute to balance it seems a sensible idea.
So, the question now is which sources of cloaking should have cloaked shield permeability, and how much.

Zitchas has already suggested here that the Drak and Pug cloaks shouldn't have any permeability.
The Arfecta's cloaking comes from the cloaking device, but we could always give it negative `cloaked shield permeability` to offset that from the cloaking device (after all, it already does this with `cloaking fuel`).

Other than those two, we currently have five distinct forms of cloak:
The cloaking device, the old Remnant cloak on the Starling and Gull, the improved Remnant cloak on the Merganser and Peregrine, the Maeri'het's cloak, and the Sasumata's cloak. 
![image](https://github.com/user-attachments/assets/728da674-6458-43b9-be7d-a88ded7ec1b0)

**Comment by Quantumshark** [comment]
Would be resolved by #11343.
